08-rails/routes-controllers.md

```diff
@@ -45,6 +45,77 @@ Let's break down this code:
   - **request method**: `get` in this example
   - **path**: what will be matched with the URL in the HTTP request, `"/books"` in this example
   - **controller and action**: defining the controller and action, split by the `#`. `"books#index"` in this case this would point to the `index` method in the `BooksController` class.
+  
+### Naming Routes 
+
+A `routes.rb` file which lists all the CRUD operations would look like this:
+
+```ruby
+# config/routes.rb
+Rails.application.routes.draw do
+
+  get "/books/new", to: "books#new", as: "new_book"
+  post "/books", to: "books#create"
+
+  get '/books/:id/edit', to: 'books#edit', as: "edit_book"
+  patch '/books/:id', to: 'books#update'
+
+  get "/books", to: "books#index"
+  get "/books/:id", to: "books#show", as: "book"
+
+  delete "/books/:id", to: "books#destroy", as: "delete_books"
+end
+```
+
+You may be saying, "Wait what's that `as: <STUFF>` part?"  This allows us to name a path.  It's very helpful to give each path for a resource, like books, a name.  The name can then be used later in your code to link actions together without having to give a hardcoded path like `'/books/new'` each time.  That way if we change a path, we only have to edit the `routes.rb` file instead of each line in our code where we use the route.  You can think of it like a constant variable referencing each path.  So `edit_book` is the name of the `/books/:id/edit` path.  
+
+You will see how these named paths are used when we delve into view helpers.
+
+You can always view the routes in your application by typing `rails routes` in the terminal.  Given the `routes.rb` file above, typing `rails routes` will result in the following:
+
+```bash
+$ rails routes
+        Prefix Verb   URI Pattern               Controller#Action
+      new_book GET    /books/new(.:format)      books#new
+         books POST   /books(.:format)          books#create
+     edit_book GET    /books/:id/edit(.:format) books#edit
+               PATCH    /books/:id(.:format)    books#update
+               GET    /books(.:format)          books#index
+          book GET    /books/:id(.:format)      books#show
+   delete_book DELETE /books/:id(.:format)      books#destroy
+```
+
+Notice that the name of each route is listed in the `Prefix` column.  Using the `rails routes` command allows you to quickly verify your routes and check the `routes.rb` file for errors.  
+
+
+### Custom Path Names
+
+We can use anything for a path name.  We could for example use:  `get "/books/:id", to: "books#show", as: "banana"`, if we wanted to have a path named `banana`.  That would not be a good name for a path, but it would function.  Rails convention suggests we name such a path `book`, but nothing stops us from using another name.  
```

Comment:
Could we add on to this that when students create additional routes beyond the standard CRUD set, they should give names to each of them and choose something appropriate based on what the route does?